ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -76,6 +76,59 @@ Tensor create_owned_tensor(
     return Tensor(std::move(storage), shape, data_type, layout, optional_tile);
 }
 
+OwnedBuffer create_owned_buffer_from_vector_of_floats(std::vector<float> &&data, DataType data_type) {
+    switch (data_type) {
+        case DataType::FLOAT32: {
+            return owned_buffer::create<float>(std::move(data));
+        }
+        case DataType::BFLOAT16: {
+            std::vector<::bfloat16> bfloat16_data(data.size());
+            std::transform(std::begin(data), std::end(data), std::begin(bfloat16_data), [](float value) {
+                return ::bfloat16(value);
+            });
+            return owned_buffer::create<::bfloat16>(std::move(bfloat16_data));
+        }
+        default: {
+            TT_THROW("Cannot create a host buffer!");
+        }
+    }
+}
+
+Tensor convert_float_vector_to_tt_tensor(
+    std::vector<float> &&data,
+    const std::array<uint32_t, 4> &shape,
+    DataType data_type,
+    Layout layout,
+    const std::optional<Device *> device,
+    const std::optional<MemoryConfig> &memory_config,
+    const std::optional<Tile> &tile) {
+    if (data_type == DataType::BFLOAT8_B || data_type == DataType::BFLOAT4_B) {
+        if (layout != Layout::TILE) {
+            log_warning(tt::LogAlways, "Tensor layout must be Layout::TILE for bfloat8_b or bfloat4_b! Tensor layout will be {} instead of the requested {}!", Layout::TILE, layout);
+        }
+        auto owned_buffer = create_owned_buffer_from_vector_of_floats(std::move(data), DataType::FLOAT32);
+        auto float_tensor = Tensor(OwnedStorage{owned_buffer}, shape, DataType::FLOAT32, Layout::ROW_MAJOR, tile).to(Layout::TILE);
```

Comment:
can we avoid creating ROW_MAJOR tensor with `tile` argument?

Reply:
Out of scope for this PR. We need make `to_layout` take in tile arg if it's ROW_MAJOR.